src/transforms/groupby.js

```diff
@@ -35,6 +35,35 @@ exports.attributes = {
             'with `x` [1, 3] and one trace with `x` [2, 4].'
         ].join(' ')
     },
+    nameformat: {
```

Comment:
@alexcjohnson I've added `Lib.templateString` that performs these substitutions. At the moment it just grabs object properties if it's simple or uses `nestedProperty` if the key is more complicated than just a plain word. For this case the only data available is `%{trace}` and `%{group}`, but it at least opens the door to a unified interface for taking this further (or worst-case-scenario it's only a couple lines of code if it's not more generally useful). Feedback welcome.